### DISCUSSION TITLE
Are installers for a framework valid formulae?

### DISCUSSION THREAD

**Discussion by Hari-07** [Homebrew/discussions#4594] [Writing Formulae/Casks]
### Output of `brew config`

```shell
not required
```

### Output of `brew doctor`

```shell
not required
```

### Description of issue

I'm working on a brew formula for flutter (A cask already exists but [the team says](https://github.com/flutter/website/pull/6263#issuecomment-959807789) since they can't ensure security, they're not considering it as an official installation method)

The current setup of flutter assumes that its written to a directory that it can modify, and edit since people can click into the source code and make modifications to their installation of the framework itself, which is a core attribute of flutter

One of the suggestions that came up during this discussion was using the brew formula for flutter, as something that installs something that can then check out the flutter repo in the user's home directory. 

Is this considered as an acceptable formulae? Going through the [acceptable formulae rules](https://docs.brew.sh/Acceptable-Formulae) I did not find any specific rule that'd make this not allowed, but it'd essentially be an installer for flutter rather than flutter itself that gets distributed through brew

If flutter itself is installed through brew is it valid for the formula to change permission on the cellar directory to which its installed so that files inside it maybe modified?

**Comment by SMillerDev** [comment]
If the installer builds from source, sure.
